roles/splunk/tasks/configure_auditd.yml

```diff
@@ -0,0 +1,45 @@
+---
+- name: "get {{ splunk_nix_user }} user uid"
+  getent:
+    database: passwd
+    key: "{{ splunk_nix_user }}"
+
+# - debug:
+#     msg: "{{ splunk_nix_user }} user id = {{ ansible_facts.getent_passwd[splunk_nix_user][1] }}"
+
+- name: auditd - set 20-splunk.rules
+  become: true
+  template:
+    src: 20-splunk.rules.j2
+    dest: /etc/audit/rules.d/20-splunk.rules
+    mode: 0600
+    owner: root
+    group: root
+  register: splunk_rule
+
+- name: Get auditd enabled level  # noqa 301 306
+  become: true
+  shell: auditctl -s | grep enabled | cut -d" " -f2
+  changed_when: false
+  check_mode: false
```

Comment:
Why is `check_mode` set to false?

Reply:
This task is just setting a variable, but ansible won't run `shell/command` tasks in check_mode. To ensure we don't get variable undefined on later tasks with `when` conditionals we have to force this task to run in `normal` and `check` modes.

>To force a task to run in normal mode and make changes to the system, even when the playbook is called with `--check`, set `check_mode: false`.   

https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_checkmode.html